src/OrderingComparer.ts

```diff
@@ -0,0 +1,146 @@
+    ofCompare((a1, a2) => compare(a2, a1))
+
+/**
+ * Given you already have an `OrderingComparer` for some type `A`, and you know how to
+ * map from some other type `B` to type `A`, you can effectively "re-use" your `OrderingComparer`
+ * for type `B`. Also referred to commonly as `contramap`, because the mapping is going
+ * from `B`&rarr;`A`, not from `A`&rarr;`B`.
+ *
+ * @category Utils
+ * @category Constructors
+ *
+ * @param known The `OrderingComparer` that you already have.
+ * @param map The function that can map from `B`&rarr;`A`.
+ *
+ * @returns A new `OrderingComparer` instance.
+ */
+const deriveFrom = <A, B>(
+    known: OrderingComparer<A>,
+    map: (b: B) => A
+): OrderingComparer<B> => ({
+    compare: (b1, b2) => (b1 === b2 ? 0 : known.compare(map(b1), map(b2))),
+})
+
+/**
+ * The default `OrderingComparer`. Converts both values to strings and does
+ * the default ASCII-based alphabetical comparison.
+ */
+const Default: OrderingComparer<never> = ofCompare((a1, a2) => {
+    const a1String: string = S.isString(a1) ? a1 : globalThis.String(a1)
+    const a2String: string = S.isString(a2) ? a2 : globalThis.String(a2)
+
+    return a1String.localeCompare(a2String) < 0 ? -1 : 1
+})
+
+/**
+ * Combine or merge multiple `OrderingComparer`s together **in a specific order**.
+ * Conceptually, this means, "Sort these values by this first, then this, then this"
+ * and so on.
+ *
+ * For example if you have an `OrderingComparer` that sorts strings alphabetically in
+ * a case-insensitive manner (say, `alphabeticalCiComparer`) and an `OrderingComparer`
+ * that sorts strings by their length (say, `lengthComparer`), you could generate a new
+ * "composite" `OrderingComparer` that sorts alphabetically (case-insensitive) **and
+ * then by** length.
+ *
+ * @example
+ * const alphabeticalCiThenLengthComparer =
+ *     OrderingComparer.getComposite(alphabeticalCiComparer, lengthComparer)
+ *
+ * @category Utils
+ *
+ * @remarks
+ * If no comparers are passed, will default to {@link Default}
+ *
+ * @returns A new `OrderingComparer` instance.
+ */
+const getComposite = <A>(
+    ...comparers: readonly OrderingComparer<A>[]
+): OrderingComparer<A> => {
+    if (comparers.length < 1) {
+        return Default
+    }
+
+    return ofCompare((a1, a2) =>
+        comparers.reduce<CompareResult>(
+            (result, nextComparer) =>
+                result !== 0 ? result : nextComparer.compare(a1, a2),
+            0
+        )
+    )
+}
+
+/**
+ * An `OrderingComparer` for the built-in `number` type. In ascending order.
+ */
+const Number: OrderingComparer<never> = ofCompare((n1, n2) => (n2 - n1 > 0 ? -1 : 1))
+
+/**
+ * An `OrderingComparer` for the built-in `string` type. Semantically equivalent
+ * to {@link Default}.
+ */
+const String: OrderingComparer<string> = Default
+
+/**
+ * An `OrderingComparer` for the built-in `date` type. In ascending order.
+ */
+const Date: OrderingComparer<Date> = deriveFrom(Number, date => date.valueOf())
```

Comment:
This guy FPs :grin:

Reply:
I figured `deriveFrom` is a bit easier to grok than `contramap`